contents/regex/Regex.md

```diff
@@ -0,0 +1,316 @@
+# Introduction to Regular Expressions
+
+Authors: Jamos Tay
+
+# Overview
+
+## What are Regular Expressions?
+
+A Regular Expression (or Regex for short) is a string of characters that define a search pattern. You can think of a regular expression as a list of descriptions that describe a string. These strings can be tested against other text formatted data to find, match and extract data.
+
+Imagine you're a witness of a robbery. How might you describe the robber to the police? You might say that the robber has a beard, brown hair and a facial scar. With this information, the police can round up everyone with beards, brown hair and scars as possible suspects. This is similar to how Regex works. You specify a search pattern using a string (e.g. It starts with 'C', ends with a `S`, and only consists of letters: `C[A-z]+S`) and regex searches a section of text for you, finding all the possible matches.
+
+## Why should I learn Regex?
+
+* **Regex is everywhere** - Regular Expressions are technically a language by themselves, but they are usually implemented within other languages. Almost any commonly used language, like Java, C++ or Python supports Regex is some fashion (A full list is available [here](http://www.regular-expressions.info/tools.html)). Even most decent IDEs or text editors support Regex in 'Find and replace' operations, such as Eclipse, Visual Studio or Notepad++, allowing non-programmers to benefit from Regex.
```

Comment:
`Almost any` -> `Almost all`.
`Even most decent IDEs or text editors` -> `In addition, many IDEs and text editors`.